test/semantics/io07.f90

```diff
@@ -0,0 +1,96 @@
+! Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+1001 format(A)
+
+     !ERROR: Format statement must be labeled
+     format(A)
+
+2001 format(3I8, 3Z8)
+2002 format(3I8, Z8)
+2003 format(  3  I  8  ,  3  Z  8  )
+2004 format(20PF10.2)
+2005 format(20P,F10.2)
+2006 format(20P7F10.2)
+2007 format(1X/)
+2008 format(/02x)
+2009 format(1x/02x)
+2010 format(2L2:)
+2011 format(:2L2)
+2012 format(2L2 : 2L2)
+
+     !ERROR: Expected ',' or ')' in format expression
+2101 format(3I83Z8, 'abc')
+
+     !ERROR: Expected ',' or ')' in format expression
+2102 format(  3  I  8  3  Z  8  )
+
+     !ERROR: Expected ',' or ')' in format expression
+2103 format(3I8 3Z8)
+
+     !ERROR: Expected ',' or ')' in format expression
+2104 format(3I8 Z8)
+
+     !ERROR: Expected ',' or ')' in format expression
+2105 format(1X3/)
+
+     !ERROR: Expected ',' or ')' in format expression
+2106 format(1X003/)
+
+     !ERROR: Expected ',' or ')' in format expression
+2107 format(3P7I2)
+
+     !ERROR: Expected ',' or ')' in format expression
+2108 format(3PI2)
+
+3001 format(*(I3))
+3002 format(5X,*(2(A)))
+
+     !ERROR: Unlimited format list must contain a data edit descriptor
+3101 format(*(X))
+
+     !ERROR: Unlimited format list must contain a data edit descriptor
+3102 format(5X,*(2(/)))
+
+     !ERROR: Unlimited format list must contain a data edit descriptor
+3103 format(5X, 'abc', *((:)))
+
+4001 format(2(X))
+
+     !ERROR: List repeat specifier must be positive
+     !ERROR: DT descriptor repeat specifier must be positive
+4101 format(0(X), 0dt)
+
+6001 format(((I0, B0)))
+
+     !ERROR: A descriptor 'w' value must be positive
```

Comment:
It would be clearer if the edit descriptors were in quotes too. Also, some messages say "descriptor" and some say "edit descriptor". I think it would be better to use "edit descriptor" everywhere.

Reply:
Done; thanks.  Doing this suggested other changes as well.